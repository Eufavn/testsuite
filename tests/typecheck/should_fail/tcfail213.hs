{-# LANGUAGE TypeFamilies, ConstraintKinds #-}
module ShouldFail where

type family F a :: Constraint

class (F a) => C a where
