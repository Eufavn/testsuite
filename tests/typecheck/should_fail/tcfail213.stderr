
tcfail213.hs:6:1:
    Illegal irreducible constraint F a
    in superclass/instance head context (Use -XUndecidableInstances to permit this)
    In the context: (F a)
    While checking the super-classes of class `C'
    In the class declaration for `C'
